### PR TITLE
Exposed HTTP resources

### PR DESCRIPTION
**Pull request by gguillemas**
Introduce a check that joins both `vulcan-exposed-http-endpoint` and `vulcan-exposed-files`.

This check implements the features of both checks, which are:

- Detection of certain resources present in a web server.
- Detection based on regex, status code or response.
- Resource database maintainable through configuration file.
- Possibility of specifying the severity of each resource being present.
- Possibility of supplying own list of resources though options.

Additionally, this check adds the following features:

- Significantly increased default resource database.
- Filtering of false positive in web servers that always return 200 OK.
- Concurrency in checking resources with multiple paths.
- Confidence score for each detected resource.
- Small testing helpers.

### CODE REVIEW

File: cmd/vulcan-exposed-http-resources/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+- If both conditions are specified, the check will consider the resource to be exposed if both of them are met.
+
+- If no conditions are specified the check will consider the resource to be exposed if the request returns any response.
+
+## Resources
+
+Resources that the check can identify as exposed are listed in the `resources.yaml` file. Any resource added to the file should have a short but clear description and should be accompanied preferably by a matching regular expression. If a reliable regular expression cannot be written, a status code should be used instead and the description should be preceded by the word "potentially"; the severity of the vulnerability should not be lowered for this reason alone.
+
+## Testing
```
Comment (manelmontilla) [inline cmd/vulcan-exposed-http-resources/README.md:36]: I like a lot the fact that there is an out of the box documented way to test resources.

File: cmd/vulcan-exposed-http-resources/main.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+		wg.Wait()
+		close(foundPathsChan)
+	}()
+
+	foundPaths := []string{}
+	for foundPath := range foundPathsChan {
+		foundPaths = append(foundPaths, foundPath)
+	}
+
+	if len(foundPaths) > 0 {
```
Comment (manelmontilla) [inline cmd/vulcan-exposed-http-resources/main.go:209]: Also not a strong opinion, but here, I would do:
```
if len(foundPaths) < 1 {
    return nil, nil
}
var severityRank string
		if httpResource.Severity != nil {
			if *httpResource.Severity > exposedVuln.Score {
				exposedVuln.Score = *httpResource.Severity
			}
			severityRank = rankSeverity(*httpResource.Severity)
		} else {
			severityRank = rankSeverity(report.SeverityThresholdHigh)
		}

		return map[string]string{
			"Score":       fmt.Sprintf("%.01f", *httpResource.Severity),
			"Severity":    severityRank,
			"Confidence":  rankConfidence(httpResource),
			"Description": httpResource.Description,
			"URL":         strings.Join(foundPaths, "\n"),
		}, nil
```

File: cmd/vulcan-exposed-http-resources/main.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+				return err
+			}
+		}
+
+		vuln, checked, err := exposedResources(logger, targetURL, resources)
+		if err != nil {
+			return err
+		}
+
+		if vuln != nil {
```
[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-exposed-http-resources/main.go]: I agree! Fixed :+1:

File: cmd/vulcan-exposed-http-resources/main.go (hunk 2 above)
Comment (gguillemas) [reply cmd/vulcan-exposed-http-resources/main.go]: I agree! Fixed :+1:

File: cmd/vulcan-exposed-http-resources/main.go [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+func checkResource(l *logrus.Entry, targetURL *url.URL, httpResource Resource) (map[string]string, error) {
+	foundPathsChan := make(chan string)
+
+	go func() {
+		limiter := rate.NewLimiter(rate.Limit(rateLimit), burst)
+		var wg sync.WaitGroup
+		for _, p := range httpResource.Paths {
+			wg.Add(1)
+			go func() {
+				err := limiter.Wait(context.Background())
```
[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-exposed-http-resources/main.go:168]: Nice catch! Done! :+1:

File: cmd/vulcan-exposed-http-resources/README.md (hunk 1 above)
Comment (gguillemas) [reply cmd/vulcan-exposed-http-resources/README.md:36]: Thanks man! I wanted to make sure anyone would have an easy way to test contributions.

**Pull request merged**